### PR TITLE
Do postprocessing in main rho process

### PR DESCRIPTION
**Pull request by noahl**
This refactor has several benefits:
  - can share utility functions between postprocessing and the rest of rho
  - can use logging from postprocessing
  - will be able to split postprocessing into different modules in the future

The motiviation for doing it right now is that it also lets us do
multiple Ansible runs in one Rho scan, for instance to find the home
directory of an EAP installation and then run a playbook that depends
on it.

This particular change does not add or remove functionality; it just
moves around code that was already in rho.

### CODE REVIEW

**Comment by chambridge** [comment]
I believe you can get rid of the following 2 lines in the `rho.spec` file with these changes:
https://github.com/quipucords/rho/blob/master/rho.spec#L50
https://github.com/quipucords/rho/blob/master/rho.spec#L64

**Comment by noahl** [comment]
@chambridge done, thank you!

File: rho/inventory_scan.py [hunk 1]
```diff
@@ -125,12 +135,85 @@ def inventory_scan(hosts_yml_path, facts_to_collect, report_path,
...
+        this_host.update(
+            postprocessing.process_jboss_eap_locate(
+                facts_to_collect, host_vars))
+        this_host.update(
+            postprocessing.process_jboss_eap_init_files(
+                facts_to_collect, host_vars))
+
+        postprocessing.handle_systemid(facts_to_collect, this_host)
+        postprocessing.handle_redhat_packages(facts_to_collect, this_host)
+        postprocessing.escape_characters(this_host)
```
Comment (chambridge) [inline rho/inventory_scan.py:195]: Seems not to be handling things correctly. Could be a change in the expected input, as I'm seeing newlines not being converted to spaces in the output.

Comment (noahl) [reply rho/inventory_scan.py:195]: The input data was unicode, where as before it was strings.

I just pushed a commit with a fix. Thank you for catching this!

**Review by chambridge — APPROVED** [review]
LGTM!

**Pull request merged**